spec/view_code_pr_checker_spec.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+require_relative 'spec_helper'
+
+module Danger
+  describe Danger::ViewCodePRChecker do
+    it 'should be a plugin' do
+      expect(described_class.new(nil)).to be_a Danger::Plugin
+    end
+
+    shared_examples 'view code without screenshots' do |modified_files|
+      before do
+        allow(@plugin.git).to receive(:modified_files).and_return(modified_files)
+      end
+
+      it 'warns when a PR with view code changes does not have screenshots' do
+        allow(@plugin.github).to receive(:pr_body).and_return('PR Body')
+
+        @plugin.view_changes_need_screenshots
+
+        expect(@dangerfile.status_report[:warnings].count).to eq 1
+      end
+
+      it 'does nothing when a PR with view code changes has screenshots' do
+        allow(@plugin.github).to receive(:pr_body).and_return('PR [![Alt text](https://myimages.com/boo)](https://digitalocean.com) Body')
+
+        @plugin.view_changes_need_screenshots
+
+        expect(@dangerfile.status_report[:warnings]).to be_empty
+      end
+    end
+
+    shared_examples 'PR without view code' do |modified_files|
+      before do
+        allow(@plugin.git).to receive(:modified_files).and_return(modified_files)
+        allow(@plugin.github).to receive(:pr_body).and_return('Body')
+        @plugin.view_changes_need_screenshots
+      end
+    
+      it 'does nothing' do
+        expect(@dangerfile.status_report[:warnings]).to be_empty
+      end
+    end
+
+    describe 'with Dangerfile' do
+      before do
+        @dangerfile = testing_dangerfile
+        @plugin = @dangerfile.view_code_pr_checker
+      end
+
+      context 'iOS PR has view code changes' do
+        include_examples 'view code without screenshots', ['TestView.swift', 'SimpleViewHelper.m']
+      end
+
+      context 'iOS PR has button changes' do
+        include_examples 'view code without screenshots', ['SimpleViewHelper.m', 'MyAwesomeButton.swift']
+      end
```

Comment:
Now that we have those nice `shared_examples` that makes it a one-liner to create new tests, I think it would be worth it, all while not adding too much, to duplicate those two tests but with ObjC.

```suggestion
      context 'iOS PR has view code changes in Swift files' do
        include_examples 'PR with view code changes', ['TestView.swift', 'SimpleViewHelper.m']
      end

      context 'iOS PR has button changes in Swift files' do
        include_examples 'PR with view code changes', ['SimpleViewHelper.m', 'MyAwesomeButton.swift']
      end

      context 'iOS PR has view code changes in ObjC files' do
        include_examples 'PR with view code changes', ['TestView.m', 'SimpleViewHelper.m']
      end

      context 'iOS PR has button changes in ObjC files' do
        include_examples 'PR with view code changes', ['SimpleViewHelper.m', 'MyAwesomeButton.m']
      end
```
And similar for Java vs Kotlin below of course.

Reply:
Actually now that I think about it, maybe it would be cleaner to have:
```ruby
shared_examples 'view_code_pr_checker` do
  it 'warns when a PR with view code changes does not have screenshots' do … end
  it 'does nothing when a PR with view code changes has screenshots' do … end
  it 'it 'does nothing when a PR does not have view code changes' do … end
end

context 'iOS PR with Swift files' do
  include_examples 'view_code_pr_checker', ['SimpleViewHelper.swift', 'MyButtonTester.swift']
end

context 'iOS PR with ObjC files' do
  include_examples 'view_code_pr_checker', ['SimpleViewHelper.m', 'MyButtonTester.m']
end

context 'Android PR with Kotlin files' do
  include_examples 'view_code_pr_checker', ['test_view.xml', 'SimpleViewHelper.kt']
end

context 'Android PR with Java files' do
  include_examples 'view_code_pr_checker', ['SimpleViewHelper.java', 'MyAwesomeButton.java']
end
```
?

Reply:
The problem grouping everything in one single shared example is that the `PR without view code` check will break when you DO have view code being changed, as warning will be generated. That's why I had to separate the examples.

But I like the idea of adding separate ObjC/Java tests!